### ISSUE TITLE
Panel header highlight obstructs panel menu.

### ISSUE THREAD

**Issue by joshhunt** [grafana/grafana#48044]
**What happened**:

When opening the panel menu on a relatively short panel, and when the panel menu extends past the panel and on top of _another_ panel, the panel header from the second panel obstructs the menu from the first panel, and prevents items from being clicked.

It appears this only happen for panels that don't have titles.

https://user-images.githubusercontent.com/46142/164439368-76fb5ec9-53b2-46f6-a057-4c6a4a7f79cc.mp4

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

1. Import the following dashboard
2. Open the panel menu on the first panel
3. Try to click "Inspect" - the panel header on the _second_ panel will be hovered instead.

<details>
<summary>Dashboard JSON</summary>

```json
{
  "__inputs": [
    {
      "name": "DS_GDEV-TESTDATA",
      "label": "gdev-testdata",
      "description": "",
      "type": "datasource",
      "pluginId": "testdata",
      "pluginName": "TestData DB"
    }
  ],
  "__elements": [],
  "__requires": [
    {
      "type": "grafana",
      "id": "grafana",
      "name": "Grafana",
      "version": "9.0.0-pre"
    },
    {
      "type": "panel",
      "id": "stat",
      "name": "Stat",
      "version": ""
    },
    {
      "type": "datasource",
      "id": "testdata",
      "name": "TestData DB",
      "version": "1.0.0"
    }
  ],
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": null,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "${DS_GDEV-TESTDATA}"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "continuous-RdYlGr"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 4,
        "w": 8,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "colorMode": "background",
        "graphMode": "area",
        "justifyMode": "auto",
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "text": {
          "titleSize": 22
        },
        "textMode": "auto"
      },
      "pluginVersion": "9.0.0-pre",
      "targets": [
        {
          "alias": "Open the menu for this panel",
          "datasource": {
            "type": "testdata",
            "uid": "${DS_GDEV-TESTDATA}"
          },
          "refId": "A",
          "scenarioId": "random_walk"
        }
      ],
      "type": "stat"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "${DS_GDEV-TESTDATA}"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "continuous-RdYlGr"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 5,
        "w": 8,
        "x": 0,
        "y": 4
      },
      "id": 3,
      "options": {
        "colorMode": "background",
        "graphMode": "area",
        "justifyMode": "auto",
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "text": {
          "titleSize": 22
        },
        "textMode": "auto"
      },
      "pluginVersion": "9.0.0-pre",
      "targets": [
        {
          "alias": "This panel should obstruct \"Inspect\" on the previous panel's menu",
          "datasource": {
            "type": "testdata",
            "uid": "${DS_GDEV-TESTDATA}"
          },
          "refId": "A",
          "scenarioId": "random_walk"
        }
      ],
      "type": "stat"
    }
  ],
  "schemaVersion": 36,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Short panels",
  "uid": "z0htbHw7z",
  "version": 4,
  "weekStart": ""
}
```

</details>

**Anything else we need to know?**:

Have a nice day :) 

**Environment**:
- Grafana version: main, `8677552dda`

**Comment by baur** [comment]
#35989

**Comment by glintik** [comment]
This bug is annoying and definitely should by fixed. PR attached.

**Comment by DmitrySidorow** [comment]
I think it was alredy fixed in 8.5.4

> This bug is annoying and definitely should by fixed. PR attached.

**Comment by natellium** [comment]
Duplicate https://github.com/grafana/grafana/issues/25930
